pkg/agent/controller/networkpolicy/reject.go

```diff
@@ -117,6 +118,12 @@ func (c *Controller) rejectRequest(pktIn *ofctrl.PacketIn) error {
 
 	sIface, srcFound := c.ifaceStore.GetInterfaceByIP(srcIP)
 	dIface, dstFound := c.ifaceStore.GetInterfaceByIP(dstIP)
+
+	isAntreaFlexibleIPAMSrc, isAntreaFlexibleIPAMDst, ctZone, dstFound, err := parseFlexibleIPAMStatus(pktIn, c.nodeConfig, srcIP, srcFound, dstIP, dstFound)
+	if err != nil {
+		return err
+	}
+
 	// isServiceTraffic checks if it's a Service traffic when the destination of the
 	// reject response is on local Node. When the destination of the reject response is
 	// remote, isServiceTraffic will always return false. Because there is no
```

Comment:
Is it safe to decide service traffic by checking dstFound and dstMAC == gwMAC? Could you check with @hongliangl @wenyingd on this?

Reply:
`AntreaProxy` is required when FlexibleIPAM is enabled, thus in this function, we are using `openflow.ServiceEPStateField` to check if the traffic is service, and this method works as expected.

Reply:
Updated comments in this part.

Reply:
I saw the case when AntreaProxy is enabled. I am asking for the case when AntreaProxy is disabled - Is it safe to decide service traffic by checking dstFound and dstMAC == gwMAC?

Reply:
A reminder for this question.

Reply:
@hongliangl @wenyingd 
I checked the service traffic decision code in AntreaProxy disabled case.
`dstIsLocal && dstMAC == gwIfaces[0].MAC.String()`
This means the reject packet will be sent to GW interface (to be De-DNATed by Node iptables), and the destination IP is local Pod.
Could you answer Jianjun's question above?

Reply:
With noEncap mode using kube-proxy, Node-to-Service traffic also uses gw as in_port. The traffic is expected to be a Service, but "dstIsLocal == false" leads to the return value is not Service?

Reply:
Thinking more, I think it is correct to get the conclusion that the traffic is for Service on K8s worker Node by checking "dstFound and dstMAC == gwMAC", with a condition that the traffic dst is a local Pod. However, the existing code has an exception that if a non-Service packet is sending to antrea-gw0 is possible to be identified as Service, because dstFound is marked as true when querying Interface by gw0 IP, and the MAC is gw's MAC.

Reply:
> Node-to-Service traffic also uses gw as in_port. The traffic is expected to be a Service, but "dstIsLocal == false because of in_port condition" leads to the return value is not Service?

Nice catch. Since FlexibleIPAM requires `AntreaProxy` but is not verified with `ProxyAll`, we can enhance document about this restriction.

Reply:
The restriction is `FlexibleIPAM requires AntreaProxy but is not verified with ProxyAll`.
Let's focus on FlexibleIPAM ACNP rejection issue in this PR and create another one for FlexibleIPAM+ProxyAll validation/support, since this issue is more urgent.

Reply:
We can do another PR, but could you create an issue to track that?

Reply:
> We can do another PR, but could you create an issue to track that?

Sure.